Homework2/DiagonalMatrix/DiagonalMatrix.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DiagonalMatrix
+{
+    internal class DiagonalMatrix
+    {
+        internal int[] diagonalNumbers { get; }
+        public int Size { get; }
+
+        public DiagonalMatrix(params int[] diagonalNumbers)
+        {
+            this.diagonalNumbers = diagonalNumbers;
+            Size = diagonalNumbers == null ? 0 : diagonalNumbers.Length;
+        }
+
+        public int this[int i, int j]
+        {
+            get
+            {
+                return i == j ? diagonalNumbers[i] : 0;
+            }
+            set
+            {
+                if (i == j)
+                {
+                    diagonalNumbers[i] = value;
+                }
+            }
+        }
+
+        public int Track()
+        {
+            int _sum = 0;
```

Comment:
pls name variables without underscore _ .
It may be applied for private/protected fields, but not for free variables. 

This is not a rule of c# of course, but just common practice.